lib/BuildSystem/BuildSystem.cpp

```diff
@@ -404,14 +405,7 @@ class TargetTask : public Task {
     auto value = BuildValue::fromData(valueData);
 
     if (value.isMissingInput()) {
-      hasMissingInput = true;
-
-      // FIXME: Design the logging and status output APIs.
-      auto& system = getBuildSystem(engine);
-      system.error(system.getMainFilename(),
-                   (Twine("missing input '") +
-                    target.getNodes()[inputID]->getName() +
-                    "' and no rule to build it"));
+      namesOfMissingInputs.push_back(target.getNodes()[inputID]->getName());
```

Comment:
We should probably just track the Node, rather than the name, and then extract the name when we want to render it to a diagnostic.